R/insertPackage.R

```diff
@@ -116,7 +116,7 @@ insertPackage <- function(file,
     if (commit) {
         if (haspkg) {
             repo <- git2r::repository(repodir)
-            setwd(pkgdir)
+            setwd(repodir)
```

Comment:
Also notice the `file.path(reldir)` add lines below, it doesn't quite match the shell commands in the non-git2r half of the function.

Reply:
Could you be more explicit, possibly with a worked example.

Burden of proof is on you.  The package works for me and other as is, and as I outlined the suggested change here is simply not good in my book and will block this PR forever unless removed.   Or you make me change my mind.

Reply:
the  logic in MWE would be equivalent to the below psuedocode:

```
if haspkg then
  cd repo/bin/windows/contrib/3.4/
  git2r add bin/windows/contrib/3.4/drat_0.1.4.zip
  git2r add bin/windows/contrib/3.4/PACKAGES
  git2r add bin/windows/contrib/3.4/PACKAGES.gz 
  git2r add bin/windows/contrib/3.4/PACKAGES.rds 
  git2r commit
else
  cd repo/bin/windows/contrib/3.4/
  git add drat_0.1.4.gzip PACKAGES PACKAGES.gz PACKAGES.rds
  git commit
end if
```
where in the git2r branch the paths are prefixed with reldir but not in the git shell branch. 

The git shell command will error (on linux) if you use the reldir prefix after cd-ing to it:

```nfultz@neal-SH97:repo$cd bin/windows/contrib/3.4
nfultz@neal-SH97:3.4$touch drat_0.1.4.zip
nfultz@neal-SH97:3.4$git add bin/windows/contrib/3.4/drat_0.1.4.zip
fatal: pathspec 'bin/windows/contrib/3.4/drat_0.1.4.zip' did not match any files```

I think the reldir prefix used to be required for git2r when those lines were written for drat, but since then git2r has added path normalization to `git2r::add()` to emulate the git shell command.  

From what I can tell from debugging, 
  ```git2r::add()``` tries to normalize the path relative to the repo root, so if you ` setwd("bin/windows")` then `git2r::add("contrib/3.4/drat_0.1.4.zip")` works correctly because git2r has R code to normalize the relative path to be relative to the repo root; however `setwd("bin/windows"); git2r::add("bin/windows/contrib/3.4/drat_0.1.4.zip")` would work on linux (and  your windows environment but not mine/appveyor).

My initial suspicion was that windows \ seperators were breaking a regex inside git2r, but that apparently is not the case in my MWE.

---

Instead of changing `setwd(pkgpath)`, would you be open to removing the reldir prefix in the `git2r::add` calls? I think that would make the two arms of the if statement more symmetric.

Reply:
I am not sure how much time and interest I have for setting up test cases.

AFAICT there is still nothing broken.  You seem annoyed that files are prefixed in one case but not the other.  I only care that the files end up in the git repo -- which appears to happen.  

So I still do not see a need for change.  I may be blind (and I am chasing another bug somewhere else but still...).